account_setup.py

```diff
@@ -0,0 +1,369 @@
+        Prompt user for confirmation, 'y' for confirm, 'n' for deny
+        :param prompt: str
+        :param deny_error: Exception
+        :param unrecognized_input_error: Exception
+        :return None if prompt successful, else raises error
+    """
+    confirmation = input(prompt).lower()
+    for i in range(3):
+        if confirmation == "n":
+            raise deny_error
+        elif confirmation == "y":
+            break
+        elif confirmation != "y" and i == 2:
+            raise unrecognized_input_error
+        confirmation = input("Please input 'y' or 'n': ").lower()
+
+
+def prompt_with_default(key, value):
+    user_value = input("{0} [{1}]: ".format(key, value))
+    if user_value != "":
+        return user_value
+    else:
+        return value
+
+
+def prompt_tenant_selection(tenant_ids):
+    print("Multiple tenants detected. Please input the ID of the tenant you wish to use.")
+    print("Tenants:", ", ".join(tenant_ids))
+    given_tenant_id = input("Please input the ID of the tenant you wish to use: ")
+    for i in range(3):
+        if given_tenant_id in tenant_ids:
+            return given_tenant_id
+        if i != 2:
+            given_tenant_id = input("Input not recognized, please try again: ")
+    raise AccountSetupError("Tenant selection not recognized after 3 attempts.")
+
+
+if __name__ == "__main__":
+    print("\nGetting credentials.")
+    # get credentials and tenant_id
+    creds, subscription_id = credentials.get_azure_cli_credentials()
+    subscription_client = SubscriptionClient(creds)
+    tenant_ids = [tenant.id for tenant in subscription_client.tenants.list()]
+    if len(tenant_ids) != 1:
+        tenant_id = prompt_tenant_selection(tenant_ids)
+    else:
+        tenant_id = tenant_ids[0]
+
+    print("Input the desired names and values for your Azure resources. "\
+          "Default values are provided in the brackets. "\
+          "Hit enter to use default.")
+    kwargs = {
+        "region": prompt_with_default("Azure Region", DefaultSettings.region),
+        "resource_group": prompt_with_default("Resource Group Name", DefaultSettings.resource_group),
+        "storage_account": prompt_with_default("Storage Account Name", DefaultSettings.storage_account),
+        "batch_account": prompt_with_default("Batch Account Name", DefaultSettings.batch_account),
+        # "virtual_network_name": prompt_with_default("Virtual Network Name", DefaultSettings.virtual_network_name),
+        # "subnet_name": prompt_with_default("Subnet Name", DefaultSettings.subnet_name),
+        "application_name": prompt_with_default("Active Directory Application Name", DefaultSettings.application_name),
+        "application_credential_name": prompt_with_default("Active Directory Application Credential Name", DefaultSettings.resource_group),
+        "service_principal": prompt_with_default("Service Principal Name", DefaultSettings.service_principal)
+    }
+    print("Creating the Azure resources.")
```

Comment:
I just went through this again and parts of this can take a while, especially getting the service principal. I would recommend adding more print statements in this piece of code make it feel like progress is being made.